OPHD/States/MapViewStateIO.cpp

```diff
@@ -450,15 +450,18 @@ void MapViewState::readStructures(Xml::XmlElement* element)
 			auto& rcc = *static_cast<RobotCommand*>(&structure);
 			XmlAttribute* robotsAttribute = structureNode->firstChildElement("robots")->firstAttribute();
 
-			for (const auto& string : NAS2D::split(robotsAttribute->value(), ','))
+			if (!robotsAttribute->value().empty())
```

Comment:
Not digging the deeper nesting here. Would rather have the code block that handles robot id lists just be skipped but I'm failing to come up with a better way to do this that doesn't use `goto` to skip it. `break` and `continue` aren't good options either.

Could just have this spit out a log message if empty? I dunno, @DanRStevens if you can think of a better way to do this let me know before merging this PR. Thanks!

Reply:
Might want to check the `firstChildElement` call is actually able to find an appropriate node before attempting to call `firstAttribute`.

From the docs, it appears `firstChildElement` can return `nullptr` if no such node is found:
http://www.grinninglizard.com/tinyxmldocs/classTiXmlHandle.html#f0aea751320f5e430fac6f8fff3b8dd4

Reply:
You're generally right, but since the save file is generated by the game, it should always be there.

Maybe for the sake of less fragile code I should put it in there, but then there's the question of silently ignore or throw?

Reply:
Well, it's an external text file. Someone can always edit it with a text editor and mess with the data. Generally best to never trust external data files that people can mess with.

Reply:
That is fair. I'll make that change here and undo the nested if since the latest change to NAS2D fixes the issue. Thanks!